### PR TITLE
Add `cuda_internal::DeviceInternals` to wrap handle etc.

### LINKED ISSUE #6819: CUDA handles are not accessible by ops
`cuda_internal::DeviceInternals device_internals_` held by `CudaDevice` which should contain all private members of `CudaDevice` needed by e.g. op implementations. A free function to obtain the device internals from a device must be provided as well.

### PR DESCRIPTION
**Pull request by asi1024**
Fix #6819.

### CODE REVIEW

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Travis @ https://travis-ci.org/chainer/chainer/builds/518128643

File: chainerx_cc/chainerx/cuda/cuda_device.h [hunk 1]
```diff
@@ -213,14 +241,9 @@ class CudaDevice : public Device {
     std::shared_ptr<MemoryPool> pinned_memory_pool_;
 
     // Memory keeper.
-    cuda_internal::MemoryKeeper memory_keeper_{};
+    cuda_internal::DeviceInternals device_internals_;
```
Comment (niboshi) [inline chainerx_cc/chainerx/cuda/cuda_device.h]: Incorrect correspondence with a comment.

File: chainerx_cc/chainerx/cuda/cuda_device.h [hunk 2]
```diff
@@ -47,14 +52,37 @@ class MemoryKeeper {
...
+    explicit DeviceInternals(int index) : index_{index}, cublas_handle_{index}, cudnn_handle_{index} {}
+
+    cuda_internal::CublasHandle& cublas_handle() { return cublas_handle_; }
 
     cuda_internal::CudnnHandle& cudnn_handle() { return cudnn_handle_; }
 
+    cuda_internal::CudaConv& cuda_conv() { return cuda_conv_; }
+
+private:
+    int index_;
```
Comment (niboshi) [inline chainerx_cc/chainerx/cuda/cuda_device.h]: Any reason to store the index?

**Comment by hvy** [comment]
Discussed offline that these comments are not critical and that they can be addressed in a separate PR. CI failures look irrelevant so let me merge this one.

File: chainerx_cc/chainerx/cuda/cuda_device.h (hunk 1 above)
Comment (hvy) [reply chainerx_cc/chainerx/cuda/cuda_device.h]: Fixed in https://github.com/chainer/chainer/pull/6827.

File: chainerx_cc/chainerx/cuda/cuda_device.h (hunk 2 above)
Comment (hvy) [reply chainerx_cc/chainerx/cuda/cuda_device.h]: Thanks, it seems not to be needed. Removed in https://github.com/chainer/chainer/pull/6827.

**Pull request merged**